src/AWS4AuthRequest.jl

```diff
@@ -75,9 +77,12 @@ function sign_aws4!(method::String,
 
     # HTTP headers...
     rmkv(headers, "Authorization")
-    setkv(headers, "x-amz-content-sha256",  content_hash)
-    setkv(headers, "x-amz-date",  datetime)
-    setkv(headers, "Content-MD5", base64encode(body_md5))
+    setkv(headers, "host", url.host)
+    setkv(headers, "x-amz-date", datetime)
+    include_md5 && setkv(headers, "Content-MD5", base64encode(body_md5))
+    if (aws_service == "s3" && method == "PUT") || include_sha256
```

Comment:
It would be good to have a reference to the S3 documentation that says this is required